scipy/integrate/_rules/_base.py

```diff
@@ -0,0 +1,657 @@
+    """
+
+    def __init__(self, base_rules):
+        for rule in base_rules:
+            if not isinstance(rule, FixedRule):
+                raise ValueError("base rules need to be instance of \
+FixedRule")
+
+        self.base_rules = base_rules
+
+    @cached_property
+    def nodes_and_weights(self):
+        nodes = _cartesian_product(
+            [rule.nodes_and_weights[0] for rule in self.base_rules]
+        )
+
+        weights = np.prod(_cartesian_product(
+            [rule.nodes_and_weights[1] for rule in self.base_rules]
+        ), axis=-1)
+
+        return nodes, weights
+
+
+class ProductNestedFixed(NestedFixedRule):
+    """
+    Find the n-dimensional cubature rule constructed from the Cartesian product of 1-D
+    `NestedFixedRule` quadrature rules, and estimate the error as the difference between
+    the product of the higher rules and the product of the lower rules.
+
+    Given a list of N 1-dimensional quadrature rules which support error estimation
+    using NestedFixedRule, this will find the N-dimensional NestedRule cubature rule
+    obtained by taking the Cartesian product of their nodes, and estimating the error by
+    taking the difference with a lower-accuracy N-dimensional cubature rule obtained
+    using the `.lower_nodes_and_weights` rule in each of the base 1-dimensional rules.
+
+    Parameters
+    ----------
+    base_rules : list of NestedRule
+        List of base 1-dimensional `NestedRule` quadrature rules.
+
+    Attributes
+    ----------
+    base_rules : list of NestedRule
+        List of base 1-dimensional `NestedRule` qudarature rules.
+
+    Examples
+    --------
+
+    Evaluate a 2D integral by taking the product of two 1D rules:
+
+    >>> import numpy as np
+    >>> from scipy.integrate import cub
+    >>> from scipy.integrate._rules import (
+    ...  ProductNestedFixed, GaussKronrodQuad
+    ... )
+    >>> def f(x):
+    ...     # f(x) = cos(x_1) + cos(x_2)
+    ...     return np.sum(np.cos(x), axis=-1)
+    >>> rule = ProductNestedFixed(
+    ...     [GaussKronrodQuad(15), GaussKronrodQuad(15)]
+    ... ) # Use 15-point Gauss-Kronrod, which implements NestedRule
+    >>> a, b = np.array([0, 0]), np.array([1, 1])
+    >>> rule.estimate(f, a, b) # True value 2*sin(1), approximately 1.6829
+     np.float64(1.682941969615793)
+    >>> rule.estimate_error(f, a, b)
+     np.float64(2.220446049250313e-16)
+    """
+
+    def __init__(self, base_rules):
+        for rule in base_rules:
+            if not isinstance(rule, NestedFixedRule):
+                raise ValueError("base rules for product need to be instance of \
+NestedFixedRule")
+
+        self.base_rules = base_rules
+
+    @cached_property
+    def nodes_and_weights(self):
+        nodes = _cartesian_product(
+            [rule.nodes_and_weights[0] for rule in self.base_rules]
+        )
+
+        weights = np.prod(_cartesian_product(
+            [rule.nodes_and_weights[1] for rule in self.base_rules]
+        ), axis=-1)
```

Comment:
this could definitely be clearer - a suggestion
```suggestion
        weights = np.prod(
            _cartesian_product(
                [rule.nodes_and_weights[1] for rule in self.base_rules]
            ),
            axis=-1,
        )
```

or perhaps clearer to put the `_cartesian_product([rule.nodes_and_weights[1] for rule in self.base_rules])` on a separate line